src/Tools/dotnet-monitor/Egress/EgressArtifactSettings.cs

```diff
@@ -3,7 +3,7 @@
 
 namespace Microsoft.Diagnostics.Tools.Monitor.Egress
 {
-    internal sealed class EgressArtifactSettings
+    public sealed class EgressArtifactSettings
```

Comment:
Does this need to be public?

Reply:
Yes and no, I initially changed this because the extension I wrote used an assembly reference to `dotnet-monitor` but I think I'm going to replace that with a source file reference, so this doesn't need to be public anymore.

Reply:
Rather than going the route of using source file references, consider making another assembly e.g. Microsoft.Diagnostics.Monitoring.Egress that can hold sharable types, even if they are still marked internal. I know that we have a bunch of collection rule code that uses source file references, but that was done so we could get the feature done in time for 6.0, but I would like to refactor those to be in common assemblies such as the existing Microsoft.Diagnostics.Monitoring.Options assembly.